walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,102 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+    // first iterate through and combine events that overlap
+    // subtract combined events calendar from whole day
+    // remove time ranges that are too short 
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+        List<TimeRange> avaliableRanges = new ArrayList<TimeRange>();
+        avaliableRanges.add(TimeRange.WHOLE_DAY);
+        Collection<TimeRange> bookedRanges = combinedMandatoryRanges(events, request);
+
+        for (TimeRange booked : bookedRanges) {
+            TimeRange lastInAvaliable = avaliableRanges.get(avaliableRanges.size()-1);
```

Comment:
Why is checking only the last available range sufficient? If it is, it would be good to leave a comment explaining this

Reply:
The bookedRanges is already sorted, so the only conflicts they may have with available ranges will be the latest one in availableRanges.

comment added.